### PR TITLE
support reshard attention mask in intermediate_api

### PR DESCRIPTION
**Pull request by liym27**
#### Before submitting

- [ ] Lint code. If there are lint issues, please format the code first.

```shell
# Install and register `pre-commit` in the project folder
pip install pre-commit && pre-commit install

# Process previous code files separately
pre-commit run --file XXXX.py
```

- [ ] Add test cases into `tests` folder. If there are codecov issues, please add tests cases first.

### CODE REVIEW

**Review by XieYunshen — APPROVED** [review]

File: paddlenlp/transformers/llama/modeling_network.py [hunk 1]
```diff
@@ -23,6 +23,9 @@
 import paddle
 import paddle.nn.functional as F
 from paddle import nn
+from paddle.distributed.auto_parallel.intermediate.tensor_parallel import (
+    PrepareLayerInput,
```
Comment (ZHUI) [inline paddlenlp/transformers/llama/modeling_network.py:27]: 什么时候加的API，要不 try catch？

**Review by ZHUI — APPROVED** [review]
LGTM
